pyproject.toml

```diff
@@ -46,6 +46,20 @@ pydocstyle = "^4.0"
 pytest = "^7.2.2"
 tox    = "^4.4.6"
 
+[tool.poetry.group.docs.dependencies]
+sphinx                           = "^5.3.0"
+sphinx-argparse                  = "^0.3.2"
+sphinx-rtd-theme-github-versions = "^1.1"
+sphinx-rtd-theme                 = "^1.2.0"
+myst-parser                      = "^1.0.0"
+sphinx-notfound-page             = "^0.8.3"
+sphinx-copybutton                = "^0.5.2"
+furo = "^2023.3.27"
+sphinxext-opengraph = "^0.8.2"
+sphinx-autobuild = "^2021.3.14"
+sphinx-design = "^0.4.1"
+sphinx-multiversion = "^0.2.4"
```

Comment:
`poetry install` now seems to install these dependencies – is that intentional? What if a user just wants to use Darglint2 and doesn't need to build the documentation?

Reply:
It shouldn't. It should only install the deps in the docs group when specifying `--with docs` which can only be used when installing from source afaik.

Reply:
Ah, I had another look into the docs. As I said the dep groups can only be installed from source using poetry. By default all dep groups are installed when installing from source. However you can declare groups to be optional development dependencies. Which groups should be declared as such in your opinion? 

You can always use the `--with` or `--without` flags to select the groups to install.

Reply:
> However you can declare groups to be optional development dependencies. Which groups should be declared as such in your opinion?

@akaihola

Reply:
>[...] optional development dependencies. Which groups should be declared as such in your opinion?

My gut answer is `dev`, `lint`, `test` and `docs`. This way a `git clone ... ; poetry install` creates a minimal footprint and uses minimal bandwidth and time, ending up with a runnable `darglint2`.

But I virtually never use `poetry` myself, so I don't actually know what users of that tool expect.

Reply:
I decided to make lint, test and docs optional since they install additional deps. 

> But I virtually never use `poetry` myself, so I don't actually know what users of that tool expect.

I am also new to poetry but I think installing with poetry is aimed at developers while others could use `pip install git@url`.